platform/ios/src/MGLAnnotationView.mm

```diff
@@ -146,6 +151,29 @@ - (void)updateScaleTransformForViewingDistance
     }
 }
 
+- (void)setRotatesToMatchCamera:(BOOL)rotatesToMatchCamera
+{
+    if (_rotatesToMatchCamera != rotatesToMatchCamera)
+    {
+        _rotatesToMatchCamera = rotatesToMatchCamera;
+        [self updateRotateTransform];
+    }
+}
+
+- (void)updateRotateTransform
```

Comment:
As far as I can tell, this method only gets called when `rotatesToMatchCamera` is set explicitly by application code. But the annotation view’s transform matrix is always overridden whenever the map viewport changes. `-updateScaleTransformForViewingDistance` gets called by `-setCenter:`, which coincidentally happens to be the method that `-[MGLMapView update AnnotationViews]` calls to ~~clobber~~ reposition the annotation view to match the map. We could merge this method and `-updateScaleTransformForViewingDistance` into an `-updateTransform` method that comes up with the canonical transform to apply.